app/script/telemetry/calling/CallLogger.js

```diff
@@ -0,0 +1,186 @@
+      type = 6;
+      charactersToKeep = 12;
+    } else {
+      // IPv4 - Without dots the maximum is 12 characters
+      type = 4;
+      charactersToKeep = 4;
+      if (ip.length > 7) {
+        charactersToKeep = 6;
+      } else if (ip.length > 6) {
+        charactersToKeep = 5;
+      }
+    }
+
+    const mid = Math.round((ip.length + 1) / 2);
+
+    return {
+      slicedIp: ip.slice(mid - charactersToKeep / 2, mid + charactersToKeep / 2),
+      type,
+    };
+  }*/
+
+  obfuscateSdp(sdpMessage, conversationId) {
+    const decodedSdpMessage = sdpTransform.parse(sdpMessage);
+
+    for (const index in decodedSdpMessage.media) {
+      // Remove fingerprint
+      if (
+        typeof decodedSdpMessage.media[index].fingerprint !== 'undefined' &&
+        typeof decodedSdpMessage.media[index].fingerprint.hash !== 'undefined'
+      ) {
+        decodedSdpMessage.media[index].fingerprint.hash =
+          'XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX';
+      }
+
+      // Remove ice password
+      if (typeof decodedSdpMessage.media[index].icePwd !== 'undefined') {
+        decodedSdpMessage.media[index].icePwd = 'X'.repeat(24);
+      }
+
+      // Remove KASE public key (for receiving side)
+      if (typeof decodedSdpMessage.media[index].invalid !== 'undefined') {
+        for (const indexInvalid in decodedSdpMessage.media[index].invalid) {
+          if (decodedSdpMessage.media[index].invalid[indexInvalid].value.startsWith('x-KASEv1')) {
+            decodedSdpMessage.media[index].invalid[indexInvalid].value = 'x-KASEv1:XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX';
+          }
+        }
+      }
+
+      // Prevent recovery of original IPs
+      // Remove bytes from the IP, concatenate it with the conversation id and the current date of the week then hash it
+      // and use that hash to derive a deterministic IP from it
+      if (typeof decodedSdpMessage.media[index].candidates !== 'undefined') {
+        for (const indexCandidate in decodedSdpMessage.media[index].candidates) {
+          decodedSdpMessage.media[index].candidates[indexCandidate].ip = this.obfuscateIp(
+            decodedSdpMessage.media[index].candidates[indexCandidate].ip
+          );
+
+          if (typeof decodedSdpMessage.media[index].candidates[indexCandidate].raddr !== 'undefined') {
+            decodedSdpMessage.media[index].candidates[indexCandidate].raddr = this.obfuscateIp(
+              decodedSdpMessage.media[index].candidates[indexCandidate].raddr
+            );
+          }
+        }
+      }
+    }
+
+    return sdpTransform.write(decodedSdpMessage);
+  }
+
+  getDebugType(number) {
+    switch (number) {
+      case 0:
+      case 300:
+      case 400:
+      case 500:
+        return 'VERBOSE';
+      case 700:
+        return 'DEBUG';
+      case 800:
+        return 'INFO';
+      case 900:
+        return 'WARNING';
+      case 100:
```

Comment:
Can we have constants for these numbers?! Maybe we can also have ranges instead of cases..

```js
if (number < DEBUG_CODE_ERROR) {
  return 'ERROR';
} else if (number < DEBUG_CODE_INFO) { 
  return 'INFO';
} else ...

```

Reply:
I don't think it's a good idea for readability to have ranges especially since you will never have a debug code 101.

Reply:
Where does 200 belong though?